test/functional/sessions.test.js

```diff
@@ -113,8 +113,10 @@ describe('Sessions', function () {
 
           return client
             .withSession(testCase.operation(client))
-            .catch(() => expect(client.topology.s.sessionPool.sessions).to.have.length(1))
-            .then(() => expect(client.topology.s.sessionPool.sessions).to.have.length(1))
+            .then(
+              () => expect(client.topology.s.sessionPool.sessions).to.have.length(1),
+              () => expect(client.topology.s.sessionPool.sessions).to.have.length(1)
+            )
```

Comment:
Is this change implying something or just seems fit b/c regardless of outcome the sessions should still be length 1?

Reply:
it's just ensuring we always check this case, no major logical difference from the code before except that this ensures we don't make this check twice in the case of an assertion failure